defaults/main.yml

```diff
@@ -1,16 +1,19 @@
-aiida_version: "1.3.0"
+aiida_version: "1.4.2"
 aiida_tag: v{{ aiida_version }}
 aiida_extras:
 - rest
 - docs
 - atomic_tools
-- testing
 - notebook
 aiida_data_folder: "${HOME}/.local/share/aiida"
 aiida_templates_folder: "${HOME}/.local/share/aiida"
 aiida_source_folder: "${HOME}/src"
 aiida_examples_folder: "${HOME}"
 aiida_venv: "${HOME}/.virtualenvs/aiida"
+aiida_constraints_file: "{{ aiida_data_folder }}/constraints.txt"
+# these are additional to those extracted from aiida-core and its extras
+aiida_constraints:
```

Comment:
Should this be removed?

Reply:
Ah, I see; we need to lock things down even further (?).
Should we be adding this constraint in the `setup.json`?

Reply:
perhaps 🤷 (to the docs extra), I guess with the new `pip --use-feature=2020-resolver` it should resolve this and not allow such a broken environment to transpire. But without it, and with this particular set of extras it definitely creates a broken environment

Reply:
@chrisjsewell that's why I advocate for having proper env requirements+pins managed by pip-tools instead of extras

Reply:
Yeh we do already have lock files for CI testing: https://github.com/aiidateam/aiida-core/tree/develop/requirements.
The problem is that we are not just trying to install aiida-core, we need to install a range of other plugin packages that might not be exactly compatible with these hard pinnings: https://github.com/marvel-nccr/ansible-role-aiida/blob/c709088dff74d1e1ae4d8379e740aba35fb2ef97/defaults/main.yml#L60-L71

Reply:
Oh, so you need a combined lockfile, I guess... Just dump that to a file and use a few requirements sources to produce transitive pins? FWIW I think what confused me is the definition substitution (constraints vs requirements).

Reply:
Yeh exactly, I'm not completely sold on the current solution here, but at least now it is properly tested and produces a valid environment, which is a step forward 😬 
So I am going to merge this as this, but anyway we have an open issue https://github.com/marvel-nccr/quantum-mobile/issues/137 to consider this further, at which point I will definitely take into account your recommendation and may well switch to some form of lock file

Reply:
FYI @webknjaz you might be interested to see I now set up a full ~2 hour vagrant+ansible build running in GH Actions: https://github.com/marvel-nccr/quantum-mobile/runs/1286287172?check_suite_focus=true 😄